internal/coreinternal/golden/normalize_timestamps.go

```diff
@@ -0,0 +1,167 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package golden // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/golden"
+
+import (
+	"sort"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func normalizeTimestamp(metrics pmetric.Metrics) {
+	zeroStartTimestamp(metrics)
+	rms := metrics.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		for j := 0; j < rms.At(i).ScopeMetrics().Len(); j++ {
+			for k := 0; k < rms.At(i).ScopeMetrics().At(j).Metrics().Len(); k++ {
+				m := rms.At(i).ScopeMetrics().At(j).Metrics().At(k)
+				switch m.Type() {
+				case pmetric.MetricTypeGauge:
+					normGaugeDataPointSlice(m.Gauge().DataPoints())
+				case pmetric.MetricTypeSum:
+					normGaugeDataPointSlice(m.Sum().DataPoints())
+				case pmetric.MetricTypeHistogram:
+					normHistogramDataPointSlice(m.Histogram().DataPoints())
+				case pmetric.MetricTypeExponentialHistogram:
+					normExponentialHistogramDataPointSlice(m.ExponentialHistogram().DataPoints())
+				case pmetric.MetricTypeSummary:
+					normSummaryDataPointSlice(m.Summary().DataPoints())
+				}
+			}
+		}
+	}
+}
+
+func zeroStartTimestamp(metrics pmetric.Metrics) {
```

Comment:
Not specifically. I started out trying to make it a bit flexible, so that if we wanted we could apply an option whether to zero the startTimestamps or not. But looking back it's probably fine to just combine these two methods into one.